[testing] move golangci-lint to github actions

thaJeztah opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah committed:
move golangci-lint to github actions

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
getting closer;

```
[/home/runner/work/docker/docker/src/github.com/docker/docker/pkg/devicemapper/devmapper_log.go:5:8: could not import C (cgo preprocessing failed)] errors in package: [/home/runner/work/docker/docker/src/github.com/docker/docker/pkg/devicemapper/devmapper_log.go:5:8: could not import C (cgo preprocessing failed)]]"
```

thaJeztah committed:
gosec: G306: Expect WriteFile permissions to be 0600 or less

    pkg/pidfile/pidfile.go:43:12: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        if err := ioutil.WriteFile(path, []byte(fmt.Sprintf("%d", os.Getpid())), 0644); err != nil {
                  ^
    pkg/plugins/pluginrpc-gen/main.go:82:33: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        errorOut("error writing file", ioutil.WriteFile(*outputFile, src, 0644))
                                       ^
    testutil/fakecontext/context.go:99:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fp, content, 0644)
               ^
    daemon/graphdriver/graphtest/graphtest_unix.go:310:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(path, data, 0700)
               ^
    layer/filestore.go:243:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fms.getMountFilename(mount, "mount-id"), []byte(mountID), 0644)
               ^
    layer/filestore.go:250:9: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        return ioutil.WriteFile(fms.getMountFilename(mount, "init-id"), []byte(init), 0644)
               ^

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

tianon commented on .github/workflows/validate.yaml:
These are probably printing `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`, so I'd suggest using `apt-get` here instead:

```suggestion
          sudo apt-get -q update
          sudo apt-get -q install \
```

tianon reviewed (commented):
Nice!  I like it :smile:

tianon commented:
Doh, managed to confuse my browser tabs and reviewed in the wrong place :facepalm:

thaJeztah committed:
golang-ci: update to 1.40.1 and update to use new config options

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
golangci.yml: replace golint with revive as it is deprecated

    WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Philetran commented:
golangci_lint_action
